### PR TITLE
regenerate font atlas when fontFamily changed

### PR DESCRIPTION
**Pull request by gnavvy**
Follow up PR for #976 

#### Change List
- regenerate font atlas when fontFamily changed

### CODE REVIEW

File: modules/core/src/core-layers/text-layer/text-layer.js [hunk 1]
```diff
@@ -54,23 +54,32 @@ const defaultProps = {
...
   initializeState() {
-    const {gl} = this.context;
-    const {fontFamily} = this.props;
-    const {mapping, texture} = makeFontAtlas(gl, fontFamily);
-    this.state = {
-      iconAtlas: texture,
-      iconMapping: mapping
-    };
+    this.state = {};
+    this.updateFontAtlas(this.props.fontFamily);
```
Comment (Pessimistress) [inline modules/core/src/core-layers/text-layer/text-layer.js]: This should be removed. `oldProps.fontFamily !== props.fontFamily` is triggered on initialization.

File: modules/core/src/core-layers/text-layer/text-layer.js [hunk 2]
```diff
@@ -54,23 +54,32 @@ const defaultProps = {
...
   }
 
   updateState({props, oldProps, changeFlags}) {
     if (
       changeFlags.dataChanged ||
+      oldProps.fontFamily !== props.fontFamily ||
```
Comment (Pessimistress) [inline modules/core/src/core-layers/text-layer/text-layer.js]: `transformStringToLetters()` only changes when text is changed, so `fontFamily` has no effect.

Instead, `MultiIconLayer` must invalidate `calculateInstanceOffsets` when `iconMapping` changes.

**Comment by Pessimistress** [comment]
I don't think MultiIconLayer properly invalidates instanceOffsets when font changes.
Can you add a font selector in layer-browser to test?

**Comment by gnavvy** [comment]
Added a control for fontFamily testing in layer-browser.
It seems only monospaced font works for now, the offset calculation for non-monospaced fonts like Helvetica is messed even before fontFamily changes, should not be the instanceOffsets invalidatoin issue.

**Pull request merged**